### PR TITLE
lint: document track docs

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Document required presence of track docs

See https://github.com/exercism/configlet/pull/389

### CODE REVIEW

**Review by ee7 — APPROVED** [review]
LGTM, although we could improve the wording at the top of the file:

> ### Rule: required files are present 
>  
> The linter should check if all the required files are present. The non-exercise specific files that must be present are: 
>  
> - `config.json` 
>
> The Concept Exercise specific files that must be present are:
> ...

since this PR adds some required files that are "non-exercise specific".

**Comment by ErikSchierboom** [comment]
Good catch! I've updated.

File: building/configlet/lint.md [hunk 1]
```diff
@@ -25,6 +25,16 @@ These are the linting rules being checked.
...
+- `docs/ABOUT.md`
+- `docs/INSTALLATION.md`
+- `docs/LEARNING.md`
+- `docs/RESOURCES.md`
+- `docs/SNIPPET.txt`
+- `docs/TESTS.md`
+- `exercises/shared/.docs/help.md`
+- `exercises/shared/.docs/tests.md`
+
+### Rule: exercises/shared/.docs/tests.md is valid
```
Comment (ee7) [inline building/configlet/lint.md]:
```suggestion
```

Copy-paste mistake? :)

**Review by ee7 — APPROVED** [review]
Non-blocking question: should we add more documentation (outside of `lint.md`) for the files in `docs/`? We used to have:
- https://github.com/exercism/legacy-docs/blob/f215e4c3f9b07f3eddd01969e62b3389dd16e97a/contributing-to-language-tracks/README.md
- https://github.com/exercism/legacy-docs/blob/f215e4c3f9b07f3eddd01969e62b3389dd16e97a/language-tracks/documentation/for-consumers.md

Currently, a reader might wonder: what's the difference between `docs/TESTS.md` and `exercises/shared/.docs/tests.md`?

The latter file is documented as:
> ## File: `tests.md`
> 
> **Purpose:** Contains track-specific instructions on how to run the tests
> 
> **Presence:** Required
> 
> Describe how to run the tests for this particular exercise.
> 
> ### Example
> 
> ```markdown
> # Tests
> 
> To run the tests, run the command `dotnet test` from within the exercise directory.
> ```

Is the intention that `TESTS.md` can/should be more detailed? That's what I'm guessing from e.g.:
- https://github.com/exercism/csharp/blob/4bacfc98a1e2d2224904fee0673ac99993a1b495/docs/TESTS.md
- https://github.com/exercism/csharp/blob/4bacfc98a1e2d2224904fee0673ac99993a1b495/exercises/shared/.docs/tests.md
- https://github.com/exercism/elixir/blob/main/docs/TESTS.md
- https://github.com/exercism/elixir/blob/main/exercises/shared/.docs/tests.md

**Comment by ErikSchierboom** [comment]
> Non-blocking question: should we add more documentation (outside of lint.md) for the files in docs/? We used to have:

We probably should.

> Is the intention that TESTS.md can/should be more detailed? That's what I'm guessing from e.g.:

That is correct

**Pull request merged**